### DISCUSSION TITLE
How to store json type of data into a single document ?

### DISCUSSION THREAD

**Discussion by kirtanchandak** [appwrite/appwrite#5704] [General]
```
const data = [
    {
      img: "https://earth5r.org/wp-conten.jpg",
      title: "Garbage collection Drive",
      date: "FriDay 9th June 2023 2:15",
      des: "Come Join us for the most beaautiful organisational comittie event :)",
    },
    {
      img: "https://www.bookeventz.com/blog/wp-content/uploads/2021/02/FI-1.jpg",
      date: "FriDay 9th June 2023 2:15",
      title: "Food Distribution Drive",
      des: "Come Join us for the most beaautiful organisational comittie event :)",
    }
  ];
```

this type of data into a single document ??

**Comment by joeyouss** [comment]
Hi, you can try to use `string` attribute once and use ` JSON.stringify()` for storing. Parsing can be done using `JSON.parse()`.
Ref: https://appwrite.io/docs/databases#attributes

**Comment by koodeau** [comment]
Is it a feature request for custom json field or just a question how to put some json data into document? If it’s a second then JSON.stringify(value) should work,
Otherwise same could be done on the backend side with some serialization, shouldn’t affect anything if used with REST API because of content-type. Anyway storing as a string and then parsing it back is a common practice, not just for backends

**Comment by step135** [comment]
Any plan to implement the support of JSON type?

**Comment by vidy01** [comment]
json.stringify doesn't work. unless i manually encode every key value pair it appwrite is throwin up.

x: {"metadata":{"components":[{"label":"Text Field","applyMaskOn":"change","tableView":true,"key":"textField","type":"textfield","input":true},{"label":"Password","applyMaskOn":"change","tableView":false,"key":"password","type":"password","input":true}]}}

z: {"metadata":"hello world"}




appwrite accepts z. but does not accept x. 

i even tried manually adding quotes around the value part of metadata, like so:
x: {"metadata":"{"components":[{"label":"Text Field","applyMaskOn":"change","tableView":true,"key":"textField","type":"textfield","input":true},{"label":"Password","applyMaskOn":"change","tableView":false,"key":"password","type":"password","input":true}]}}"

thoughts?
